### PR TITLE
networkmanager: add openvpn interface handler

### PR DESCRIPTION
**Pull request by artur-rs**
Patch changes:
    - added OpenVPN settings dialog
    - implemented TLS authority
    - implemented certficate handlers

### CODE REVIEW

**Comment by mvollmer** [comment]
@artur-rs, thanks!

**Comment by mvollmer** [comment]
@artur-rd, how would we test this as part of our integration tests?

Can you give a quick example of setting up a typical OpenVPN connection, including the server side?

**Comment by mvollmer** [comment]
One needs to have the `network-manager-openvpn` package installed for this.  We probably should use `pkg/cockpit-component-install-dialog` to let the user install it on-demand.

**Comment by mvollmer** [comment]
There are many changes here besides adding support for OpenVPN, such as switching the "Status" and "Carrier" lines around. There might be good reasons for that switch and the other changes, but let's try to keep them separate.

**Comment by artur-rs** [comment]
> There are many changes here besides adding support for OpenVPN, such as switching the "Status" and "Carrier" lines around. There might be good reasons for that switch and the other changes, but let's try to keep them separate.

@mvollmer as I mentioned [here](https://github.com/cockpit-project/cockpit/pull/12498#issuecomment-519529803), each new patch contains the [general changes](https://github.com/cockpit-project/cockpit/pull/12526/files) - I thought that this approach would be better for anyone willing for building/manual testing. All changes regarding open-vpn are located in the [2nd commit](https://github.com/cockpit-project/cockpit/pull/12529/commits/3978b3b80316f6b91cdcc6d9359aae736c0e6261)

File: pkg/networkmanager/index.html [hunk 1]
```diff
@@ -107,19 +107,29 @@ <h2 class="panel-title">
...
+                <li><a tabindex="0" id="networking-add-vlan" translatable="yes">VLAN</a></li>
+              </ul>
+          </div>
         </div>
       </div>
       <table class="table table-hover">
         <thead>
           <tr>
             <th translatable="yes">Name</th>
+            <th translatable="yes">Interface</th>
```
Comment (mvollmer) [inline pkg/networkmanager/index.html]: We don't want to expose the connection id, so just `Name` should be enough. But I can see how you want VPN configs to have more descriptibe names than just "vpn0" or worse, "tun0"... Hmm.

File: pkg/networkmanager/index.html [hunk 2]
```diff
@@ -107,19 +107,29 @@ <h2 class="panel-title">
...
+            <th translatable="yes">Type</th>
             <th translatable="yes">IP Address</th>
-            <th translatable="yes" class="networking-speed">Sending</th>
-            <th translatable="yes" class="networking-speed">Receiving</th>
+            <th translatable="yes" class="networking-speed">Status: (Sending/Receiving)</th>
```
Comment (mvollmer) [inline pkg/networkmanager/index.html]: This is ugly, IMO, with lots of punctuation and jumping text when the IO rates change.  I'd keep the current design and make the column smaller if necessary.

File: pkg/networkmanager/index.html [hunk 3]
```diff
@@ -135,9 +145,10 @@ <h2 class="panel-title" translatable="yes">Unmanaged Interfaces</h2>
...
```
Comment (mvollmer) [inline pkg/networkmanager/index.html]: See above.

File: pkg/networkmanager/index.html [hunk 4]
```diff
@@ -135,9 +145,10 @@ <h2 class="panel-title" translatable="yes">Unmanaged Interfaces</h2>
         <thead>
           <tr>
             <th translatable="yes">Name</th>
+            <th translatable="yes">Interface</th>
```
Comment (mvollmer) [inline pkg/networkmanager/index.html]: See above.

**Comment by mvollmer** [comment]
> @mvollmer as I mentioned [here](https://github.com/cockpit-project/cockpit/pull/12498#issuecomment-519529803), each new patch contains the [general changes](https://github.com/cockpit-project/cockpit/pull/12526/files) - I thought that this approach would be better for anyone willing for building/manual testing.

Ahh, damn, I missed that you have a separate PR just for the first commit... this makes a lot of sense, of course.

I will copy my review comments over to #12526. :-P

**Comment by mvollmer** [comment]
> I will copy my review comments over to #12526. :-P

Done!

File: pkg/networkmanager/interfaces.js [hunk 5]
```diff
@@ -2919,17 +3022,18 @@ PageNetworkInterface.prototype = {
...
-                render_ip_settings_row("ipv6", _("IPv6")),
+            return [ render_interface_section_separator("Settings"),
+                render_master(),
+                render_general_settings_row(),
                 render_mtu_settings_row(),
                 render_vlan_settings_row(),
                 render_bridge_settings_row(),
                 render_bridge_port_settings_row(),
                 render_bond_settings_row(),
                 render_team_settings_row(),
```
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Shouldn't there be a "VPN Settings" row?

**Comment by andreasn** [comment]
Right now it looks like "Import a saved VPN configuration" is a child of "Device private key". Is this correct?
Shouldnt it be connected to the "Import (.ovpn)" label?

Also, if I import, does that mean I will still need to set the gateway, the different certificates and stuff, or does those things have nothing to do with each other?

**Comment by mvollmer** [comment]
Please reopen if you want to still work with us on this.

**Comment by Elxafil** [comment]
Hey is it possible to get this continued? I love the project and given the support for multiple machines within the same cockpit panel it would be handy to have openvpn/ipsec capabilities for secure load balancing and trafficing.

**Pull request closed without merging**